kubetest/azure_helpers.go

```diff
@@ -282,6 +296,31 @@ func (az *AzureClient) DeleteResourceGroup(ctx context.Context, groupName string
 	return nil
 }
 
+func (az *AzureClient) AssignOwnerRoleToIdentity(ctx context.Context, resourceGroupName, identityName string) error {
+	identity, err := az.msiClient.Get(ctx, resourceGroupName, identityName)
+	if err != nil {
+		return fmt.Errorf("failed to get identity's client ID: %v", err)
+	}
+
+	identityPrincipalID := identity.PrincipalID.String()
+	// Grant the identity 'Owner' access to the subscription
+	// so it can pull images from private registry
+	roleDefinitionID := fmt.Sprintf(aadRoleReferenceTemplate, az.subscriptionID, aadOwnerRoleID)
+	scope := fmt.Sprintf(aadRoleResourceGroupScopeTemplate, az.subscriptionID)
+	roleAssignmentParameters := authorization.RoleAssignmentCreateParameters{
+		Properties: &authorization.RoleAssignmentProperties{
+			RoleDefinitionID: stringPointer(roleDefinitionID),
+			PrincipalID:      stringPointer(identityPrincipalID),
+		},
+	}
+
+	if _, err := az.authorizationClient.Create(ctx, scope, uuid.NewV1().String(), roleAssignmentParameters); err != nil {
```

Comment:
have we checked our e2e credentials have the permission to do the role assignment?

Reply:
I assigned 'Owner' role to the service principal in prow.